src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1679,9 +1681,14 @@ private function moveNode(string $srcAbsPath, string $destAbsPath): void
             throw new PathNotFoundException("Parent of the destination path '".$destAbsPath."' has to exist.");
         }
 
-        $query = 'SELECT path, id FROM phpcr_nodes WHERE path LIKE ? OR path = ? AND workspace_name = ? '.$this->getConnection()
-                ->getDatabasePlatform()->getForUpdateSQL()
-        ;
+        $forUpdateSql = ' FOR UPDATE'; // https://github.com/doctrine/dbal/blob/79ea9d6eda8e8e8705f2db58439e9934d8c769da/src/Platforms/AbstractPlatform.php#L1776
+        if ($this->getConnection()->getDatabasePlatform() instanceof SqlitePlatform) {
+            $forUpdateSql = ''; // https://github.com/doctrine/dbal/blob/79ea9d6eda8e8e8705f2db58439e9934d8c769da/src/Platforms/SqlitePlatform.php#L753
+        } elseif ($this->getConnection()->getDatabasePlatform() instanceof SQLServerPlatform) {
+            $forUpdateSql = ''; // https://github.com/doctrine/dbal/blob/79ea9d6eda8e8e8705f2db58439e9934d8c769da/src/Platforms/SQLServerPlatform.php#L1625
+        }
+
+        $query = 'SELECT path, id FROM phpcr_nodes WHERE path LIKE ? OR path = ? AND workspace_name = ? '.$forUpdateSql;
```

Comment:
Have you tried leveraging the query builder for adding lock hints?

Reply:
Not familiar with that. @dbu how would you change this part?

Reply:
i am also not familiar with the dbal query builder. did doctrine drop the getForUpdateSQL method? does the changelog not explain what to do instead?

Reply:
> does the changelog not explain what to do instead?

https://github.com/doctrine/dbal/blob/79ea9d6eda8e8e8705f2db58439e9934d8c769da/UPGRADE.md?plain=1#L11-L14